metabase/metabase
# [WIP] Change `getMetadata` selector to compute data based on the RTK query cache only

Closes #43630

:warning: Do not merge. It's out of draft only to run CI :warning: 

### TODO

- [x] adopt approach from #42066
- [x] verify that it fixes the simple case
- [x] cover all table-providing endpoints
  - [x] `listDatabaseSchemaTables`
  - [x] `/api/table/:id` -> table only
  - [x] `/api/table`
  - [x] `/api/table/:id/query_metadata`
  - [x] `/api/dataset/query_metadata` -> databases, tables, fields
  - [x] `/api/card/:id/query_metadata` -> databases, tables, fields
  - [x] `/api/dashboard/:id/query_metadata` -> databases, tables, fields, cards
  - [x] ~reusable util?~
  - [x] go over all `provide*Tags` to see if anything's missing
- [x] try without database-providing endpoints covered
  - [x] either cover all database-providing endpoints, or start a new PR or...
  - [ ] then decide for fields
- [ ] fields
  - [ ] `/api/field/:id` -> field only
- [ ] cards
  - [ ] `/api/card/:id` -> card only
- [x] use raw table fields in `hydrateTableFields`
- [x] `mergeEntities`
- [ ] `state: any`
- [ ] things don't work
  - [x] browse/databases
  - [ ] everything disappears after 1min
    - [ ] `metabase-api/queries/removeQueryResult` action
- [ ] add test for basic case
  - [ ] maybe it already exists, find it
- [ ] what about `uploadManagementApi.endpoints.listUploadTables` (it's EE-only)?
- [ ] mock rtk query reducer(s) in unit tests
- [x] use endpoint timestamps to choose which data to use

### Description

TODO

### How to verify

1. Create a native model, `select 1 as A, 2 as B, 3 as C`, run it, save it
2. Edit model's metadata
3. Map all 3 fields to the same db field (e.g. Orders > Id)
4. Give fields unique names to make it easy to spot the issue (e.g. ID1, ID2, ID3)
5. Save metadata
6. Open notebook
7. Click dropdown near data source table
8. Verify that you can select from 3 different columns: [ID1, ID2, ID3](https://github.com/metabase/metabase/assets/6830683/8d9ebc35-0387-4c47-8e02-18e4f489991e) (in `master` it's [ID3, ID3, ID3](https://github.com/metabase/metabase/assets/6830683/663b61eb-f614-4709-9db5-43acac3f1d3f))
